packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -3122,6 +3145,7 @@ function commitReconciliationEffects(
   const flags = finishedWork.flags;
   if (flags & Placement) {
     commitHostPlacement(finishedWork);
+    commitFragmentInstanceInsertionEffects(finishedWork);
```

Comment:
You assume that `finishedWork` is a HostComponent here but it might not be.

E.g. `<Fragment ref={...}>{c ? <Component><div /></Component> : null}</Fragment>`.

In this case it would be the `<Component>` fiber that's being placed.

Instead, you should do this in `insertOrAppendPlacementNode` and `insertOrAppendPlacementNodeIntoContainer`. Those are recursively traversing down to find all the HostComponents inside the placement.

Reply:
One reason I put it here is because `commitHostPlacement` is gated by `!supportsMutation`. We eventually want this all to work in Fabric. Though there are probably additional considerations to make with that, I haven't taken a closer look at the RN implementation yet.

Reply:
Moved into `insertOrAppendPlacementNode/IntoContainer`. Will revisit when its time to enable for Fabric.